debezium/src/main/scala/akka/stream/alpakka/debezium/scaladsl/DebeziumSource.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.debezium.scaladsl
+
+import java.util
+import java.util.concurrent.{ExecutorService, Executors}
+
+import akka.NotUsed
+import akka.actor.{ActorSystem, Props}
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.debezium.impl.{CommitterActor, LazyEmbeddedEngine}
+import akka.stream.alpakka.debezium.{CommittableRecord, DebeziumSettings, SimpleRecord}
+import akka.stream.scaladsl._
+import akka.stream.{OverflowStrategy, QueueOfferResult}
+import io.debezium.embedded.Connect
+import io.debezium.engine.DebeziumEngine
+import org.apache.kafka.connect.source.SourceRecord
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Await, ExecutionContext, Future, Promise}
+
+object DebeziumSource {
+  def custom(settings: DebeziumSettings)(implicit s: ActorSystem): Source[CommittableRecord, NotUsed] = {
+    val configurations = settings.properties
+
+    val valueConverter = settings.internalValueConverter.getDeclaredConstructor().newInstance()
+    valueConverter.configure(configurations.asInstanceOf[java.util.Map[String, String]], false)
+
+    val engine = new LazyEmbeddedEngine.BuilderImpl().using(configurations).build()
+
+    lazy val committer = engine.committer()
+
+    val singleThreadEc = ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor())
+
+    Source
+      .repeat(engine)
+      .mapAsync(1) { engine =>
```

Comment:
AFAIK, Debezium polling is single threaded most time, but this parameter could be customised I guess.